apps/console/src/features/server-configurations/utils/governance-connector-utils.ts

```diff
@@ -86,4 +85,91 @@ export class GovernanceConnectorUtils {
     public static decodeConnectorPropertyName(name: string): string {
         return name.split("-").join(".");
     }
+
+    /**
+     * Governance connectors and their sub categories that will be visible in a sub organization
+     */
+    public static readonly SHOW_GOVERNANCE_CONNECTORS_FOR_SUBORGS = [
+        {
+            connectors: [
+                {
+                    friendlyName: "Account Recovery",
+                    id: "YWNjb3VudC1yZWNvdmVyeQ",
+                    name: "account-recovery",
+                    properties: [
+                        "Recovery.Notification.Password.Enable", // Notification based password recovery
+                        "Recovery.Notification.Username.Enable", // Username recovery
+                        "Recovery.Notification.InternallyManage" // Manage notifications sending internally
+                    ]
+                }
+            ],
+            id: "QWNjb3VudCBNYW5hZ2VtZW50",
+            name: "Account Management"
+        }
+    ]
+
+    /**
+     * Filter governance categories of a connector for a sub organization.
+     * @param governanceConnectors - List of categories to evaluate.
+     * @param governanceCategoryId - Category id of the governance connector.
+     * 
+     * @returns Filtered categories as a list.
+     */
+    public static filterGovernanceConnectorCategories
+    (governanceCategoryId: string, governanceConnectors: any[]): any[] {
+        let showGovernanceConnectors = [];
+
+        for (let i = 0; i < this.SHOW_GOVERNANCE_CONNECTORS_FOR_SUBORGS.length; i++) {
+            if (governanceCategoryId === this.SHOW_GOVERNANCE_CONNECTORS_FOR_SUBORGS[i].id) {
+
+                showGovernanceConnectors = this.SHOW_GOVERNANCE_CONNECTORS_FOR_SUBORGS[i].connectors;
+
+                break;
+            }
+        }
+
+        const showGovernanceConnectorsIdOfSuborgs = [];
+
+        showGovernanceConnectors.forEach(connector => {
+            showGovernanceConnectorsIdOfSuborgs.push(connector.id);
+        });
+
+        for (let i = governanceConnectors.length - 1; i >= 0; i--) {
+            if (!showGovernanceConnectorsIdOfSuborgs.includes(governanceConnectors[i].id)) {
+                governanceConnectors.splice(i, 1);
+            } else {
+                const showProperties = this.getGovernanceConnectorsProperties(showGovernanceConnectors,
+                    governanceConnectors[i].id);
+
+                for (let j = governanceConnectors[i].properties.length - 1; j >= 0; j--) {
+                    if (!showProperties.includes(governanceConnectors[i].properties[j].name)) {
+                        governanceConnectors[i].properties.splice(j, 1);
+                    }
+                }
+            }
+        }
```

Comment:
I think we can merge these multiple iterations to reduce the number of iterations.